### PR TITLE
MAINT: Enforce string type for where parameter

### PR DESCRIPTION
**Pull request by gfyoung**
Deprecated in 0.11.0.

xref #12027.

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. can you add a deprecation removal note. ping when pushed.

**Comment by jreback** [comment]
also this *might* still be used in the doc-build (though I think I took most of this out). can you check?

**Comment by gfyoung** [comment]
@jreback : I don't see any AFAICT.  FYI, going to add a modified version of test I deleted to ensure that we raise when non-strings are passed in.

File: pandas/computation/pytables.py [hunk 1]
```diff
@@ -452,6 +450,26 @@ def _rewrite_membership_op(self, node, left, right):
...
+
+    Raises
+    ------
+    TypeError : An invalid data type was passed in for w (e.g. dict).
+    """
+
+    if not (isinstance(w, (Expr, string_types)) or is_list_like(w)):
+        raise TypeError("where must be passed as a string, Expr, "
+                        "or list-like of Exprs")
+
```
Comment (jreback) [inline pandas/computation/pytables.py]: you could return w here
makes it a bit more natural to use it

File: pandas/computation/pytables.py [hunk 2]
```diff
@@ -481,11 +499,9 @@ class Expr(expr.Expr):
     "major_axis>=20130101"
     """
 
-    def __init__(self, where, op=None, value=None, queryables=None,
-                 encoding=None, scope_level=0):
+    def __init__(self, where, queryables=None, encoding=None, scope_level=0):
 
-        # try to be back compat
-        where = self.parse_back_compat(where, op, value)
+        _validate_where(where)
```
Comment (jreback) [inline pandas/computation/pytables.py]: where = _validate_where(where)

**Review by jreback — APPROVED** [review]
lgtm minor comment

File: pandas/computation/pytables.py (hunk 1 above)
Comment (gfyoung) [inline pandas/computation/pytables.py]: Fair enough.  Done.

File: pandas/computation/pytables.py (hunk 2 above)
Comment (gfyoung) [inline pandas/computation/pytables.py]: Fair enough.  Done.

**Comment by jreback** [comment]
thanks @gfyoung

**Pull request closed without merging**